x-pack/plugins/cases/public/components/all_cases/all_cases_list.tsx

```diff
@@ -141,32 +140,23 @@ export const AllCasesList = React.memo<AllCasesListProps>(
             perPage: page.size,
           };
         }
-        setQueryParams(newQueryParams);
+        setUrlQueryParams(newQueryParams);
         deselectCases();
       },
-      [queryParams, deselectCases, setQueryParams]
+      [queryParams, deselectCases, setUrlQueryParams]
     );
 
     const onFilterChangedCallback = useCallback(
       (newFilterOptions: Partial<FilterOptions>) => {
         if (newFilterOptions.status && newFilterOptions.status === CaseStatuses.closed) {
-          setQueryParams((prevQueryParams) => ({
-            ...prevQueryParams,
-            sortField: SortFieldCase.closedAt,
-          }));
+          setUrlQueryParams({ sortField: SortFieldCase.closedAt });
         } else if (newFilterOptions.status && newFilterOptions.status === CaseStatuses.open) {
-          setQueryParams((prevQueryParams) => ({
-            ...prevQueryParams,
-            sortField: SortFieldCase.createdAt,
-          }));
+          setUrlQueryParams({ sortField: SortFieldCase.createdAt });
```

Comment:
Unrelated to your changes but I think we could combine these last two else-if:

```
else if (newFilterOptions.status && (newFilterOptions.status === CaseStatuses.open || newFilterOptions.status === CaseStatuses['in-progress'])) {
  setUrlQueryParams({ sortField: SortFieldCase.createdAt });
} 
```

Reply:
I had to do some testing yesterday on this callback and was thinking just the same.

Will do 👍